src/20230925_AaveV2_Eth_TUSDOffboardingPlanPartII/AaveV2_Ethereum_TUSDOffboardingPlanPartII_20230925.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {IERC20} from 'solidity-utils/contracts/oz-common/interfaces/IERC20.sol';
+import {AaveV2Ethereum, AaveV2EthereumAssets, ILendingPoolConfigurator} from 'aave-address-book/AaveV2Ethereum.sol';
+import {AaveV2PayloadEthereum, IEngine, EngineFlags} from 'lib/aave-helpers/src/v2-config-engine/AaveV2PayloadEthereum.sol';
+import {IV2RateStrategyFactory} from 'lib/aave-helpers/src/v2-config-engine/IV2RateStrategyFactory.sol';
+
+/**
+ * @title TUSD Offboarding Plan Part II
+ * @author Marc Zeller - Aave-Chan Initiative
+ * - Snapshot: https://snapshot.org/#/aave.eth/proposal/0x95cca29a9cdcaf51bb7331a9516d643a5c88f8ddce86c5f3920c2ae4d604193f
+ * - Discussion: https://governance.aave.com/t/arfc-tusd-offboarding-plan-part-ii/14863
+ */
+contract AaveV2_Ethereum_TUSDOffboardingPlanPartII_20230925 is AaveV2PayloadEthereum {
+  function _preExecute() internal override {
+    uint256 aBUSDBalance = IERC20(AaveV2EthereumAssets.BUSD_A_TOKEN).balanceOf(
+      address(AaveV2Ethereum.COLLECTOR)
+    );
+    uint256 availableBUSD = IERC20(AaveV2EthereumAssets.BUSD_UNDERLYING).balanceOf(
+      AaveV2EthereumAssets.BUSD_A_TOKEN
+    );
+    AaveV2Ethereum.COLLECTOR.transfer(
+      AaveV2EthereumAssets.BUSD_A_TOKEN,
+      address(this),
+      aBUSDBalance > availableBUSD ? availableBUSD : aBUSDBalance
+    );
+    AaveV2Ethereum.POOL.withdraw(
+      AaveV2EthereumAssets.BUSD_UNDERLYING,
+      type(uint256).max,
+      address(AaveV2Ethereum.COLLECTOR)
+    );
```

Comment:
I would recommend to deduplicate this code, and wrap it into function

Reply:
fixed by @JosepBove